build.sh

```diff
@@ -101,16 +101,26 @@ fi
 
 pushd "$BUILD_DIR"
 
-# avoid the command failed and exits
-# and cmake will check some directories to determine whether some targets built
-make clean || true
-rm -rf external/arrow-install
+if [ ! -z "$RAY_USE_CMAKE" ] ; then
+  # avoid the command failed and exits
+  # and cmake will check some directories to determine whether some targets built
+  make clean || true
+  rm -rf external/arrow-install
 
-cmake -DCMAKE_BUILD_TYPE=$CBUILD_TYPE \
-      -DCMAKE_RAY_LANG_JAVA=$RAY_BUILD_JAVA \
-      -DCMAKE_RAY_LANG_PYTHON=$RAY_BUILD_PYTHON \
-      -DRAY_USE_NEW_GCS=$RAY_USE_NEW_GCS \
-      -DPYTHON_EXECUTABLE:FILEPATH=$PYTHON_EXECUTABLE $ROOT_DIR
+  cmake -DCMAKE_BUILD_TYPE=$CBUILD_TYPE \
+        -DCMAKE_RAY_LANG_JAVA=$RAY_BUILD_JAVA \
+        -DCMAKE_RAY_LANG_PYTHON=$RAY_BUILD_PYTHON \
+        -DRAY_USE_NEW_GCS=$RAY_USE_NEW_GCS \
+        -DPYTHON_EXECUTABLE:FILEPATH=$PYTHON_EXECUTABLE $ROOT_DIR
+
+  make
+else
+  # The following line installs pyarrow from S3, these wheels have been
+  # generated from https://github.com/ray-project/arrow-build from
+  # the commit listed in the command.
+  pip install --target=$ROOT_DIR/python/ray/pyarrow_files pyarrow \
```

Comment:
What does this do? We don't want to actually "install pyarrow" when building Ray, right? We just want to "get" the wheel, right?

Reply:
Yeah, this is only used for downloading. It's the only version I have found that doesn't require a bunch if ifs for each different python version. Let me know if you find a better way (it's not that bad and doesn't actually install it to the python path).

Reply:
I see, then that's probably what we want to do with modin also, right? cc @devin-petersohn